### ISSUE TITLE
lack of demo

### ISSUE THREAD

**Issue by ghostchan** [AmazingDreams/vue-matomo#9]
lack of demo

**Comment by AmazingDreams** [comment]
There isn't a whole lot to demo, unless you take into account all the matomo operations. Besides you'd need to run a demo matomo server.

What do you propose to be included in this demo?

**Comment by siulca** [comment]
It does appear very simple but I can't get it to work and a basic demo would be proof that it works.

**Comment by lucasdcrk** [comment]
Can't get it work neither, a demo would be nice!

**Comment by AmazingDreams** [comment]
Demo can be found here: https://github.com/AmazingDreams/vue-matomo/tree/master/demo

**Comment by lucasdcrk** [comment]
The demo should show how to interact with the api 
I was able to get tracking working but not access matomo api to set user name, etc

**Comment by AmazingDreams** [comment]
In you can use `this.$matomo.setUserId(uid)` in your components. As if you're using `window.Piwik.getTracker().setUserId(uid)`. You can for instance in your main `App` component do this:

    mounted () {
         const uid = getAuthUserId()
         this.$matomo.setUserId(uid)
    }

I'll set up some more demo's

**Comment by RFSwanepoel** [comment]
I can't even get the this.$matomo to show up in the vue VM.  I can see that window.Piwik and window._paq.push exists but nothing in the vue VM.

**Comment by RFSwanepoel** [comment]
> 0.3.x

Yeah, I'm on 0.3.2 , I am trying to do a localhost setup though, not sure if this might be causing the issue? 
My app runs on https://localhost:8080 and my Matomo is running on apache at 
http://localhost/dir/another/dir 

I do get visist stats when I refresh but that's all I get fr now.

And here is my vue config:
Vue.use(VueMatomo, {
    // Configure your matomo server and site
    host: 'http://localhost/www/piwik',
    siteId: 1,
    // Enables automatically registering pageviews on the router
    router: router,
    // Require consent before sending tracking information to matomo
    // Default: false
    requireConsent: false,
    // Whether to track the initial page view
    // Default: true
    trackInitialView: true,
    // Changes the default .js and .php endpoint's filename
    // Default: 'piwik'
    trackerFileName: 'piwik'
  });

**Comment by AmazingDreams** [comment]
There may be a delay when the matomo JS has been loaded. Could you try doing something using `setTimeout` with say 1000ms delay?

**Comment by RFSwanepoel** [comment]
@AmazingDreams, Thank you. That seems to have fixed my "undefined".

**Comment by AmazingDreams** [comment]
@RFSwanepoel I have decided to include piwik.js directly. If you update to v0.4.0 it does not happen anymore.

**Comment by Joe-Wilson-Barantum** [comment]
setUserId(uid) is not working to set user id, pls advice thanks
